### PR TITLE
Various improvements suggested by Clippy.

### PR DESCRIPTION
**Pull request by ltratt**
And turn off some of the more annoying ones that would make the code more complex for no gain...

### CODE REVIEW

File: src/tester.rs [hunk 1]
```diff
@@ -207,7 +208,7 @@ impl<'a> LangTester<'a> {
             for (cmd_name, mut cmd) in cmd_pairs {
                 let output = cmd
                     .output()
-                    .expect(&format!("Couldn't run command {:?}.", cmd));
+                    .unwrap_or_else(|_| panic!(format!("Couldn't run command {:?}.", cmd)));
```
Comment (jacob-hughes) [inline src/tester.rs]: Why is this better?

Comment (vext01) [reply src/tester.rs]: I also prefer how it was before :P

File: src/parser.rs [hunk 2]
```diff
@@ -77,15 +77,15 @@ pub(crate) fn parse_tests<'a>(test_str: &'a str) -> HashMap<String, Test<'a>> {
     tests
 }
 
-fn indent_level(lines: &Vec<&str>, line_off: usize) -> usize {
+fn indent_level(lines: &[&str], line_off: usize) -> usize {
```
Comment (jacob-hughes) [inline src/parser.rs]: I've seen in the past Rust people stating that passing a slice over a `Vec` here more idiomatic. I can't seem to find the find the a link to the reasoning anymore though. Do you know why this is better?

File: src/tester.rs (hunk 1 above)
Comment (ltratt) [reply src/tester.rs]: It means that the `format` stuff isn't executed *unless* there's been a failure.

File: src/parser.rs (hunk 2 above)
Comment (ltratt) [reply src/parser.rs]: It's more flexible: you can do things like `&vec[n..m]` and so on. Is it useful here? Not really. But it's not unuseful, and it's slightly fewer characters ;)

File: src/tester.rs (hunk 1 above)
Comment (jacob-hughes) [reply src/tester.rs]: Ah. That's beautifully non-obvious. I'd expect expect to do that.

Comment (ltratt) [reply src/tester.rs]: `expect` is a normal function, so arguments are eagerly evaluated. We can't blame `expect` here :)

Comment (vext01) [reply src/tester.rs]: Interesting! In that case, I retract my comment.

Shame it makes the code less readable.

File: src/parser.rs (hunk 2 above)
Comment (vext01) [reply src/parser.rs]: I guess a slice is more generic? It could be from a Vec or and array?

Comment (jacob-hughes) [reply src/parser.rs]: Ok I'm convinced.

File: src/tester.rs (hunk 1 above)
Comment (jacob-hughes) [reply src/tester.rs]: I see that now yeah, that's why it takes a closure.

**Comment by vext01** [comment]
Yes, sorry, go ahead!

**Pull request merged**